main.go

```diff
@@ -137,11 +159,20 @@ func main() {
 		case <-ticker:
 			fmt.Printf("%v runs so far, %v failures, over %s\n",
 				runs, fails, roundToSeconds(time.Since(startTime)))
+		case <-ctx.Done():
+			// A context timeout in this case is indicative of no failures
+			// being detected in the allotted duration.
+			if err := ctx.Err(); err != context.DeadlineExceeded {
+				return err
+			}
+			return nil
```

Comment:
Now that I think about it: the value of `ctx.Err()` is irrelevant, isn't it? I.e., stress should FAIL if `fails > 0` when you get here, or SUCCEED otherwise.